### PR TITLE
Build default URL from kattisrc

### PR DESCRIPTION
**Pull request by davad**

### CODE REVIEW

File: generators/app/index.js [hunk 1]
```diff
@@ -69,9 +80,24 @@ class functionGenerator extends generator {
       );
     }
 
+    // Build default URL from kattisrc
+    // Check the same locations the official submit.py checks
+    const pathsToCheck = [
+      '/usr/local/etc/kattisrc',
+      os.homedir() + '/.kattisrc',
+      this.destinationPath('.kattisrc'),
```
Comment (karpikpl) [inline generators/app/index.js:89]: I make an assumption that .kattisrc is in ../ (one level up)

File: generators/app/index.js [hunk 2]
```diff
@@ -69,9 +80,24 @@ class functionGenerator extends generator {
...
+    ).lastIndexOf(true);
+    const config = new ConfigParser();
+    if (kattisrcFoundId > -1) {
+      config.read(pathsToCheck[kattisrcFoundId]);
+    }
+    const hostname = config.get('kattis', 'hostname') || 'open.kattis.com';
     const url =
       this.answers.url ||
-      `https://open.kattis.com/problems/${this.answers.problem}/file/statement/samples.zip`;
+      `https://${hostname}/problems/${this.answers.problem}/file/statement/samples.zip`;
```
Comment (karpikpl) [inline generators/app/index.js:101]: I don't think that's going to work, authorization is required

**Review by karpikpl — COMMENTED** [review]
I've left two comments about the URL and auth

File: generators/app/index.js (hunk 1 above)
Comment (davad) [reply generators/app/index.js:89]: I'll add that assumption too. These three are the places that `submit.py` checks.

File: generators/app/index.js (hunk 2 above)
Comment (davad) [reply generators/app/index.js:101]: I took a closer look at `submit.py`. It's making a POST to the login URL defined in the kattisrc file. Then it's using the cookies from that response in future requests.

**Pull request merged**